### ISSUE TITLE
Add a make install target to install into a prefix

### ISSUE THREAD

**Issue by wolfv** [certik/terminal#40]
... or use CMake? 
That would make it easy to have terminal on conda-forge, and we could use it right away in mamba.

I just tested it and it looks very impressive and also very much like what we need so I am happy to help out here.

**Comment by certik** [comment]
Yes, of course. So far I just copied the header files to my projects but I agree, let's just install it.

**Comment by wolfv** [comment]
@certik I am looking again at this to properly handle the yes/no prompts in mamba :) especially CTRL+C during a wait for `y`/`N` isn't handled correctly in mamba right now. Once you have the installation sorted out we can make a nice conda package and I can use it right away.

**Comment by certik** [comment]
So Ctrl-C it normally handled by the system I believe, and there is a way to turn it off, I think this line does it:

https://github.com/certik/terminal/blob/69ee07e5aee2fe4c4bff4fa164364ec049c66069/terminal_base.h#L106

If you are ready to use it, I can get it done today. I'll work on this.

**Comment by certik** [comment]
CMake build system introduced in #41.

**Comment by certik** [comment]
This is now fixed in master.

I am thinking there is no need for multiple header files. I think this library even when fully developed can fit into just one header file. Let's call it `terminal.h`.

Should it go into a subdirectory such as `terminal/terminal.h`, or directly as `terminal.h`? What is the standard convention for C++?

**Comment by wolfv** [comment]
I believe a subdirectory might be nice to not clutter the `include` folder, and not have potential conflicts with other packages. `terminal` is a pretty generic term I guess. might be worth it to use `cpp-terminal/terminal.hpp`?

Once you tag a release I am happy to make the conda package :)

**Comment by JohanMabille** [comment]
I like `cpp-terminal`, that would be consistent with other conda packages such as `cpp-filesystem`

**Comment by certik** [comment]
Perfect. I moved it under `cpp-terminal` in #43.

I still have to create a CMake package to install so that other projects can easily depend on this. Otherwise it's ready.

**Comment by certik** [comment]
The CMake package was created in #45 and #46.

**Comment by certik** [comment]
@wolfv here is the tag: https://github.com/certik/terminal/releases/tag/v0.1.0

Please keep me updated and open issues for any problems you discover.
